lib/ast-input-transform.js

```diff
@@ -1,9 +1,89 @@
 'use strict';
 
+const checkAttributes = require('./helpers/check-attributes');
+const supportsAttribute = require('./helpers/supports-attribute');
+
 const attributeToPropertyMap = {
   role: 'ariaRole',
 };
 
+const commonAttributes = [
+  'id',
+  'class',
+  'role',
+
+  // TextSupportMixin
+  'autocapitalize',
+  'autocorrect',
+  'autofocus',
+  'disabled',
+  'form',
+  'maxlength',
+  'minlength',
+  'placeholder',
+  'readonly',
+  'required',
+  'selectionDirection',
+  'spellcheck',
+  'tabindex',
+  'title',
+
+  /data-test-.+/, // support for ember-test-selectors
+];
+
+const inputAttributes = [
+  'accept',
+  'autocomplete',
+  'autosave',
+  'dir',
+  'formaction',
+  'formenctype',
+  'formmethod',
+  'formnovalidate',
+  'formtarget',
+  'height',
+  'inputmode',
+  'lang',
+  'list',
+  'type',
+  'max',
+  'min',
+  'multiple',
+  'name',
+  'pattern',
+  'size',
+  'step',
+  'value',
+  'width',
+].concat(commonAttributes);
+
+const textareaAttributes = [
+  'rows',
+  'cols',
+  'name',
+  'selectionEnd',
+  'selectionStart',
+  'autocomplete',
+  'wrap',
+  'lang',
+  'dir',
+  'value',
+].concat(commonAttributes);
+
+const checkboxAttributes = [
+  'type',
+  'checked',
+  'indeterminate',
+  'disabled',
+  'tabindex',
+  'name',
+  'autofocus',
+  'required',
+  'form',
+].concat(commonAttributes);
+
+('use strict');
```

Comment:
This confuses me, doesn't the top level `'use strict';` cover us here?

Reply:
Oh, don't know where this came from, probably my IDE messed this up. Removed.